app/models/snack.rb

```diff
@@ -0,0 +1,11 @@
+class Snack < ApplicationRecord
+  validates_presence_of :name, :price
+  validates_numericality_of :price
+
+  has_many :machine_snacks
+  has_many :machines, through: :machine_snacks
+
+  def self.average_price
+    average(:price)
+  end
```

Comment:
To answer your confusion that you mentioned in your PR comment, it does not matter if you were to do a class or an instance method. I think sometimes it is preferred to do class methods so that you can limit the amount of chained method calls within the view. Both would work perfectly fine though.